recipes/folly/all/conanfile.py

```diff
@@ -19,216 +19,241 @@ class FollyConan(ConanFile):
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        # deps.set_property("backtrace", "cmake_file_name", "Backtrace")
+        deps.set_property("boost", "cmake_file_name", "Boost")
+        deps.set_property("bzip2", "cmake_file_name", "BZip2")
+        deps.set_property("double-conversion", "cmake_file_name", "DoubleConversion")
+        deps.set_property("fmt", "cmake_file_name", "fmt")
+        deps.set_property("gflags", "cmake_file_name", "Gflags")
+        deps.set_property("glog", "cmake_file_name", "Glog")
+        # deps.set_property("libaio", "cmake_file_name", "LibAIO")
+        deps.set_property("libdwarf", "cmake_file_name", "LibDwarf")
+        deps.set_property("libevent", "cmake_file_name", "LibEvent")
+        deps.set_property("libiberty", "cmake_file_name", "Libiberty")
+        deps.set_property("libsodium", "cmake_file_name", "Libsodium")
+        deps.set_property("libunwind", "cmake_file_name", "LibUnwind")
+        # deps.set_property("liburing", "cmake_file_name", "LibUring")
+        deps.set_property("lz4", "cmake_file_name", "LZ4")
+        deps.set_property("openssl", "cmake_file_name", "OpenSSL")
+        deps.set_property("snappy", "cmake_file_name", "Snappy")
+        deps.set_property("xz_utils", "cmake_file_name", "LibLZMA")
+        deps.set_property("zlib", "cmake_file_name", "ZLIB")
+        deps.set_property("zstd", "cmake_file_name", "Zstd")
+        deps.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        folly_deps = os.path.join(self.source_folder, "CMake", "folly-deps.cmake")
+        replace_in_file(self, folly_deps, " MODULE", " ")
+        replace_in_file(self, folly_deps, "OpenSSL 1.1.1", "OpenSSL")
+        # Disable example
+        save(self, os.path.join(self.source_folder, "folly", "logging", "example", "CMakeLists.txt"), "")
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        files.rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
-        files.rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "folly")
         self.cpp_info.set_property("cmake_target_name", "Folly::folly")
```

Comment:
This target name clashes with `libfolly` component. In my opinion this one should be all lowercase

```suggestion
        self.cpp_info.set_property("cmake_target_name", "folly::folly")
```